src/wp-includes/script-loader.php

```diff
@@ -34,6 +34,39 @@
 /** ClassicPress Styles Functions */
 require( ABSPATH . WPINC . '/functions.wp-styles.php' );
 
+/**
+ * Returns a cache buster string for an enqueued script or stylesheet.
+ *
+ * @since 1.0.0-beta2
+ *
+ * @param string $type    The type of asset being enqueued ('script' or 'style').
+ * @param string $handle  The handle of the asset being enqueued (or `null` to
+ *                        get the default asset version).
+ * @return string         The cache buster string for this asset.
+ */
+function classicpress_asset_version( $type = 'script', $handle = null ) {
+	/**
+	 * The default asset version.  In the source repository this is based on
+	 * the ClassicPress version; at build time it is replaced with a random
+	 * string.
+	 */
+	$version = 'cb' . substr( md5( classicpress_version() ), 0, 6 );
+
+	/**
+	 * Allows modifying the asset version for each script and style.
+	 *
+	 * @since 1.0.0-beta2
+	 *
+	 * @param string $version The default version for this asset.
+	 * @param string $type    The type of asset being enqueued ('script' or 'style').
+	 * @param string $handle  The handle of the asset being enqueued (or `null` to
+	 *                        get the default asset version).
+	 */
+	$version = apply_filters( 'classicpress_asset_version', $version, $type, $handle );
```

Comment:
We need to document somewhere the filters/hooks that we are adding

Reply:
If I've read this correctly, we're now computing the `md5` every time a script or stylesheet is enqueued - that seems daft.

Reply:
we are computing the md5 based on cp version but as you can see on line 97 this value is cached

Reply:
No, the md5 is computed once or twice on each page load, and only when ClassicPress is running from a source install.